MinSwap.java

```diff
@@ -35,7 +35,35 @@ private static void printArray(int a[]){
     }
     private static int minSwaps(int a[]){
         //WRITE YOUR CODE HERE...
+        int p1=0, p2 = a.length - 1, temp,swap=0;
+        while(p1 < p2){
+            if (a[p1] == 0){
+                if(a[p2]  != 0){
+                    temp = a[p1];
+                    a[p1] = a[p2];
+                    a[p2] = temp;
+                    p1 ++;
+                    p2 --;
+                    swap ++;
+                }
+                else{
+                    p2 --;
+                }
+            }
+            else if (a[p1] != 0){
```

Comment:
Not an issue, but, there is no need to check  "else if (a[p1] != 0) { }" ... just use "else { }"